app/assets/javascripts/course/assessment/submission/answer/programming.js

```diff
@@ -389,30 +391,23 @@
   }
 
   /**
-   * Handles the annotation reply button click event.
+   * Handles the annotation topic reply button click event.
+   *
+   * TODO Trigger replying to last post instead of topic itself once proper behavior for post
+   *      deletion has been implemented.
    *
    * @param e The event object.
    */
-  function onAnnotationReply(e) {
+  function onAnnotationTopicReply(e) {
     var $element = $(e.target);
-    var $post = $element.parents('.discussion_post:first');
-    var $replies = $post.next('div.replies');
-
-    var courseId = courseIdForElement($element);
-    var assessmentId = assessmentIdForElement($element);
-    var submissionId = submissionIdForElement($element);
-    var answerId = answerIdForRow($element);
-    var programmingFileId = programmingFileIdForRow($element);
-    var lineNumber = $element.parents('.line-annotation:first').data('lineNumber');
-    var postId = $post.data('postId');
-
-    var $form = findOrCreateAnnotationForm($replies, courseId, assessmentId, submissionId, answerId,
-                                           programmingFileId, lineNumber, postId);
-    $form.find('textarea').focus();
+    var $addAnnotationButton =
+      $element.parents('.line-annotation:first').parents('tr:first').prev('tr').
+               find('.add-annotation');
+    $addAnnotationButton.click();
```

Comment:
Is this because you want to always display comment button in the bottom ? 

Yah a bit hackish, I think you can directly add a form there ? That's the logic in the onClick callback.